File: frontend/project/src/application/dto/transaction/TransactionDTO.ts
/** Wire shape for a ledger record (TransactionDto). */
export interface TransactionDTO {
  id: string;
  type: string | null;
  amount: number;
  description: string | null;
  walletId: string;
  relatedWalletId: string | null;
  reference: string | null;
  createdAt: string;
  isCompleted: boolean;
}

File: backend/WalletSystem.Application/Common/Dtos/TransactionDto.cs
namespace WalletSystem.Application.Common.Dtos;

public record TransactionDto(
    Guid Id,
    string Type,
    decimal Amount,
    string? Description,
    Guid WalletId,
    Guid? RelatedWalletId,
    string? Reference,
    DateTime CreatedAt,
    bool IsCompleted
);
